tests/v2/validation/pipeline/releaseupgrade/main.go

```diff
@@ -311,3 +325,35 @@ func NewRancherClusterConfiguration(cluster pipeline.RancherCluster, newConfigNa
 
 	return
 }
+
+//	NewRancherLocalClusterConfiguration is a function that accepts single cluster from the cluster matrix's local field.
+//
+// Writes a new configuration file with the original configuration's content and updates the new file's content with the given cluster values.
+func NewRancherLocalClusterConfiguration(cluster pipeline.RancherCluster, newConfigName file.Name, copiedConfig []byte) (err error) {
+	_, err = newConfigName.NewFile(copiedConfig)
+	if err != nil {
+		logrus.Info("error while writing populated config", err)
+		return err
+	}
+
+	err = newConfigName.SetEnvironmentKey(config.ConfigEnvironmentKey)
+	if err != nil {
+		logrus.Info("error while setting new config as env var", err)
+		return err
+	}
+
+	clusterID := "local"
+	upgradeConfig := new(upgradeinput.Config)
+	config.LoadAndUpdateConfig(upgradeinput.ConfigurationFileKey, upgradeConfig, func() {
```

Comment:
I see this will `panic()`, but since that doesn't return an err, does the formatting not look weird if this section fails/panics?

Reply:
Panic returns an error on the system level. So it'll do exit on the container this is running and gonna seem failure at this stage on the pipelines